uxarray/grid.py

```diff
@@ -371,3 +374,92 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
         integral = np.dot(face_areas, face_vals)
 
         return integral
+
+    def compute_antimeridian_faces(self, threshold=180.0):
+        """Locates any face that crosses the antimeridian'.
+
+        Returns
+        -------
+        crossed_indices : tuple
+            Indices into Mesh2_face_nodes corresponding to any face that
+            crossed the antimeridian
+
+        Example
+        -------
+        Given a face with (x) values
+            [a, b, c, d]
+
+        A face can be constructed with sides
+            ab, bc, cd, da
+
+        Pad each face
+            [a, b, c, d, a]
+
+        Magnitude of each side
+            abs([a-b, b-c, c-d, d-a])
+
+        Any value > threshold crosses the antimeridian
+        """
+
+        # longitude (x) values of each face
+        face_nodes = self.Mesh2_face_nodes.astype(np.int32).values
+        face_lon = self.Mesh2_node_x.values[face_nodes]
+
+        # pad last value to properly represent each side
+        face_lon_pad = np.pad(face_lon, (0, 1), 'wrap')[:-1]
+
+        # magnitude of each side
+        side_diff = np.abs(np.diff(face_lon_pad))
+
+        # any row with a single side > 180 set to true
+        crossed_mask = np.any(side_diff > threshold, axis=1)
+
+        # indices of faces that cross antimeridian
+        crossed_indices = np.argwhere(crossed_mask)
+
+        return tuple(crossed_indices)
+
+    def split_antimeridian_faces(self):
+        """Split any face that crosses the antimeridian into two new faces
```

Comment:
I noticed that the way you treated the wrap-around face is by clipping it with the 0 longitude, which is a little bit different that the way we treated the the latlonbound function.

Me and @paullric both thinks that we should constrain each edge's length by 180 degrees. And if this direction exceeds the 180 degree, we will represent the edge from another direction.

e.g. edge a --> b, but lon_b - lon_a > 180, so this edge is actually equivalent to  b --> a.

for detailed implementation, you can view #55 `uxarray/helpers.py/def insert_pt_in_latlonbox` and `uxarray/helpers.py/def get_latlonbox_width` or another algorithm used for the test cases `uxarray/helpers.py/minmax_Longitude_rad` and ` uxarray/helpers.py/expand_longitude_rad`

Reply:
Thank you for this! I put the PR back on draft earlier last week since I need to make some changes. I'll make some changes early next week. Have a great winter break!